amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/RecordsPublisher.java

```diff
@@ -47,4 +46,12 @@
      * Shutdowns the publisher. Once this method returns the publisher should no longer provide any records.
      */
     void shutdown();
+
+    /**
+     * Notify the publisher on receipt of a data event.
+     * @param ack acknowledgement received from the subscriber.
+     */
+    default void notify(RecordsDeliveryAck ack) {
+        throw new UnsupportedOperationException("RecordsPublisher does not support acknowledgement from Subscriber");
```

Comment:
> For my understanding: why is the desired default behavior to throw an exception here?
I think this is because the logic for evictAckedEventAndScheduleNextEvent is implemented in FanOutRecordsPublisher, which extends this class. FanOutRecordsPublisher is the publisher that we are using and requires the Ack mechanism.

Reply:
This is to enforce, any Publisher allowing NotifyingSubscriber to subscribe to itself, to implement the notify() method.